hugo/content/docs/dependency_injection.md

```diff
@@ -0,0 +1,91 @@
+---
+title: 'Sprotty configuration and dependency injection'
+Weight: 300
+--- 
+
+As seen in the [getting started](../getting_started) guide sprotty relies heavily on dependency injection (DI) through [InversifyJs](https://inversify.io/) for configuration of its varous components. This chapter will take a closer look at how to work with this.
+
+## Why dependency injection
+DI allows us to
+- not care about the instantiation and life-cycle of service components,
+- manage singletons like the various registries without using the global scope,
+- easily mock components in tests,
+- exchange default implementations with custom ones with minimum code changes,
+- modularize the configuration of specific features and scenarios and merge these modules for the final application
+
+## The container
+The DI-container is the main point of configuration. The standard in sprotty is naming the file containing the container `di.config.ts`.
+
+```typescript
+export const createContainer = (containerId: string) => {
+    const myModule = new ContainerModule((bind, unbind, isBound, rebind) => {
+        bind(TYPES.ModelSource).to(LocalModelSource).inSingletonScope();
+        const context = { bind, unbind, isBound, rebind };
+        configureModelElement(context, 'graph', SGraph, SGraphView);
+        configureModelElement(context, 'task', SNode, TaskNodeView);
+        configureModelElement(context, 'edge', SEdge, PolylineEdgeView);
+        configureViewerOptions(context, {
+            needsClientLayout: false,
+            baseDiv: containerId
+        });
+    });
+    const container = new Container();
+    loadDefaultModules(container);
+    container.load(myModule, edgeIntersectionModule);
+    return container;
+};
+```
+The container is build from multiple modules. Through `loadDefaultModules()` all modules are loaded for default sprotty functionality. we can of course also load other non default modules like the `edgeIntersectionModule` for extra functionality.
+
+Most important is our own module where the core of the configuration happens. Here we can configure singleton scope classes like our [datasource](../datasources) or rebind default default sprotty components like for example the logger to a custom implementation. We use Symbols for bindings instead of using classes directly. All Symbols can be found in the `TYPES` object.
```

Comment:
```suggestion
Most important is our own module where the core of the configuration happens. Here we can configure singleton scope classes like our [datasource](../datasources) or rebind default sprotty components, for example the logger, to a custom implementation. We use Symbols for bindings instead of using classes directly. All Symbols can be found in the `TYPES` object.
```